src/GUI/Buttons/Keyboard.elm

```diff
@@ -0,0 +1,258 @@
+interpret keyCode =
+    case String.toList keyCode of
+        'D' :: 'i' :: 'g' :: 'i' :: 't' :: c :: [] ->
+            -- DigitX
+            Character c
+
+        'K' :: 'e' :: 'y' :: c :: [] ->
+            -- KeyX
+            Character c
+
+        'N' :: 'u' :: 'm' :: 'p' :: 'a' :: 'd' :: c :: [] ->
+            -- NumpadX, where X is presumably a digit
+            NumpadChar c
+
+        _ ->
+            parseFKey keyCode |> Maybe.withDefault (misc keyCode)
+
+
+parseFKey : String -> Maybe Interpretation
+parseFKey keyCode =
+    case String.uncons keyCode of
+        Just ( 'F', n ) ->
+            -- FX, where X is a presumably positive integer
+            String.toInt n |> Maybe.map FKey
+
+        _ ->
+            Nothing
+
+
+{-| Miscellaneous keys.
+
+Note: No button description in the original game exceeds 7 characters in length, and 8 is the maximum that's guaranteed to fit to the left of "READY".
+
+-}
+misc : String -> Interpretation
+misc keyCode =
+    case keyCode of
+        "AltLeft" ->
+            Named "L.Alt"
+
+        "AltRight" ->
+            -- Nicely dual to "L.Alt"; they are in turn dual to "L.Ctrl" (from the original game) and "R.Ctrl".
+            Named "R.Alt"
+
+        "ArrowDown" ->
+            -- From the original game.
+            Named "D.Arrow"
+
+        "ArrowLeft" ->
+            -- From the original game.
+            Named "L.Arrow"
+
+        "ArrowRight" ->
+            Named "R.Arrow"
+
+        "ArrowUp" ->
+            Named "U.Arrow"
+
+        "AudioVolumeDown" ->
+            Named "VolDn"
+
+        "AudioVolumeMute" ->
+            Named "Mute"
+
+        "AudioVolumeUp" ->
+            Named "VolUp"
+
+        "Backquote" ->
+            Character '`'
+
+        "Backslash" ->
+            Character '\\'
+
+        "Backspace" ->
+            Named "Bksp"
+
+        "BracketLeft" ->
+            Character '['
+
+        "BracketRight" ->
+            Character ']'
+
+        "CapsLock" ->
+            Named "CapsLk"
+
+        "Comma" ->
+            -- From the original game.
+            Character ','
+
+        "ContextMenu" ->
+            Named "Menu"
+
+        "ControlLeft" ->
+            Named "L.Ctrl"
+
+        "ControlRight" ->
+            Named "R.Ctrl"
+
+        "Delete" ->
+            Named "Del"
+
+        "End" ->
+            Named "End"
+
+        "Enter" ->
+            Named "Enter"
+
+        "Equal" ->
+            Character '='
+
+        "Escape" ->
+            Named "Esc"
+
+        "Home" ->
+            Named "Home"
+
+        "Insert" ->
+            Named "Ins"
+
+        "IntlBackslash" ->
+            -- Unclear what this key can be said to represent, but it produces a '\' in the English (UK) layout.
+            Character '\\'
+
+        "Minus" ->
+            Character '-'
+
+        "NumLock" ->
+            Named "NumLk"
+
+        "NumpadAdd" ->
+            -- 'NumpadMultiply' is rendered as '*' in the original game.
```

Comment:
This comment is repeated for a few different Numpad keys. I don’t understand the comment.

Reply:
Should hopefully be clearer now.